Fix issues found by PVS-Studio static analyzer #4

dimhotepus opened the pull request:
https://pvs-studio.com/en/order/open-source-license/

dimhotepus committed:
rtspserver: Do not assign variable to itself

dimhotepus committed:
rtspserver: Drop duplicated condition check

dimhotepus committed:
rtspserver: Throw OverflowException when any

dimhotepus committed:
rtspserver: Dispose loggers after usage

dimhotepus committed:
rtspserver: Safe events invocations

https://pvs-studio.com/en/docs/warnings/v3083/print

dimhotepus committed:
rtspserver: Drop unused var

dimhotepus committed:
rtspserver: Dispose IDisposable members

dimhotepus committed:
rtspserver: Use pattern matching to speedup packets writing

dimhotepus committed:
rtsp: Use short-circuit || operator where possible

dimhotepus committed:
rtptools: Correctly build csrc argument

RtpSend.HexFormat expects 3 args, 4 were given.

juliusfriedman commented on RtspServer/RtspServer.cs:
```diff
@@ -814,10 +814,8 @@ public void EnableUnreliableTransport(int port = RtspMessage.UnreliableTransport
-                    EndPoint inBound;
-
                     m_UdpServerSocket = new Socket(addressFamily, SocketType.Dgram, ProtocolType.Udp);
-                    m_UdpServerSocket.Bind(inBound = new IPEndPoint(Media.Common.Extensions.Socket.SocketExtensions.GetFirstUnicastIPAddress(addressFamily), port));
```
inBound is not used?

juliusfriedman commented on RtpTools/RtpSend.cs:
```diff
@@ -649,19 +649,10 @@ public static string ToTextualConvention(FileFormat format, Rtp.RtpPacket packet
                 using (Media.RFC3550.SourceList sl = new Media.RFC3550.SourceList(packet))
                 {
-
-                    if (sl == null)
```
I think we should leave this but use sl.IsComplete

juliusfriedman replied:
Please put the if else back but instead of checking null use IsComplete.

dimhotepus committed:
rtp: Correctly store socket error. On linux ErrorCode != SocketErrorCode

https://blog.jetbrains.com/dotnet/2020/04/27/socket-error-codes-depend-runtime-operating-system/

juliusfriedman reviewed (approved):
Thank you for your time and contributions!!!
